### PR TITLE
Backport of PR #12569

### PR DESCRIPTION
**Pull request by PoisonPhang**
https://app.clickup.com/t/358r3ft

Backport of PR [#12569](https://github.com/paritytech/substrate/pull/12569)

Auto format did indeed auto-format

### CODE REVIEW

**Review by vimukthi-git — APPROVED** [review]
LGTM

File: frame/balances/src/tests.rs [hunk 1]
```diff
@@ -1263,5 +1263,163 @@ macro_rules! decl_tests {
...
+
+		#[test]
+		fn fungible_unbalanced_trait_decrease_balance_works() {
+			<$ext_builder>::default().build().execute_with(|| {
+				assert_ok!(<Balances as fungible::Unbalanced<_>>::set_balance(&1337, 100));
+				assert_eq!(<Balances as fungible::Inspect<_>>::balance(&1337), 100);
+
+				assert_noop!(
+					<Balances as fungible::Unbalanced<_>>::decrease_balance(&1337, 101),
+					TokenError::NoFunds
```
Comment (PoisonPhang) [inline frame/balances/src/tests.rs:1317]: Need to import this

**Comment by PoisonPhang** [comment]
I can build this branch fine, but running `cargo test --all` produces this result
```
error[E0437]: type `Event` is not a member of trait `pallet_asset_tx_payment::Config`
   --> frame/transaction-payment/asset-tx-payment/src/tests.rs:201:2
    |
201 |     type Event = Event;
    |     ^^^^^^^^^^^^^^^^^^^ not a member of trait `pallet_asset_tx_payment::Config`

For more information about this error, try `rustc --explain E0437`.
```
Can anyone else confirm this behavior?

**Comment by dzmitry-lahoda** [comment]
imho read the nix. do what nix does. nixify your local scripts and share with us. ci = local.

**Comment by PoisonPhang** [comment]
All tests pass with `cargo nextest run --workflow`, however, the same is not true with `cargo test --all`. This issue exists within the `ComposableFi/substrate/polkadot-v0.9.27` branch as well. This is not present on `paritytech/substrate` from what me and @benluelo can tell

> imho read the nix. do what nix does. nixify your local scripts and share with us. ci = local.

I need to re-partition my root directory before I can correctly run nix commands.

**Pull request merged**